ExtractorUtils/ExtractionRun.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Cognite.Extensions;
+using CogniteSdk;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Cognite.Extractor.Utils
+{
+    /// <summary>
+    /// Configuration for extraction pipeline run
+    /// </summary>
+    public class ExtractionRunConfig
+    {
+        /// <summary>
+        /// ExternalId of extraction pipeline.
+        /// </summary>
+        public string PipelineId { get; set; }
+        /// <summary>
+        /// Frequency of extraction pipeline updates in seconds.
+        /// </summary>
+        public int Frequency { get; set; }
```

Comment:
I don't think we need to have frequency configurable, it can just be hardcoded to something sensible, like 10 mins.

Reply:
I'm trying to make something generic. I'd rather have it configurable, to be honest. There might be more than one use case, and keeping it configurable does not hurt. I can remove it from the example config if needed.

Reply:
I will add a default value, however.

Reply:
Leaving parameters confugrable when they don't need to be is an anti-pattern though. It will for sure lead to users putting in weird numbers that don't make much sense, such as either leaving it very course and then complain about extraction pipelines not working, or leaving it to fine and filling up the backend DB with bloating data. 

The granularity for the not-seen notifications is expected to be 1h in the backend, so anything finer than 10 mins is not really necessary anyway.

Reply:
I did not know that. That makes this largely useless for OPC-UA and PI and other real-time data applications.

Reply:
Why?

This is mainly intended for major outages, in which 1h is plenty of granularity based on our user research.

Reply:
The issue is that this is _worse_ than the existing solutions. You are asking users to downgrade from grafana which is fast, to this which is extremely slow in comparison. If the data is lost _forever_ on an outage (which may be the case for OPC-UA), then this is just not good enough.

Reply:
No, we are not asking users to downgrade at all. 

The current way most extractor crashes is discovered is either:

 * with the data quality monitoring tool, which still in beta, and often with a granularity of up to 6-12h
 * by manually discovering a gap in data in some time series

this would be far superior to both of these. 

As far as I know there's only one deployment of OPC UA that actually uses grafana/alert manager for automatic error reporting, that's not the default way most do this. And metrics are not going away, this is a supplement. 

The notifications will include webhooks medium term, and will for example be used to automatically open tickets with our and/or the customer's ticketing systems, automatically call incidents, etc. I have been pushing for this granularity to be as low as possible, but both since the severity is meant to be high for these scenarios, false positives is not something we want. All of the users we have interviewed have said 6h is fine. Maybe we will decrease the granularity to 30 or 15 mins in the future, but for now it looks like it's going to be 1h. 

It's also worth noting that this is only for the cases the extractor did not report a failure itself, so in the majority of cases this would not be needed anyway as the crash would have been caught and a failed run would be reported in a `finally` block. Calling it useless is inaccurate.